test/test_library/unit/test_files.py

```diff
@@ -0,0 +1,511 @@
+        self.assertIsNone(f1.update(records))
+
+    # .query tests
+    # TODO: Implement.
+    # TODO: Do we really need .query property? Check for usage.
+
+    # .ref tests
+    def test_returns_files_instance_with_query_filtered_by_given_ref(self):
+        # create many files
+        self._assert_filtered_by('ref', ['ref1', 'ref2'], filter_by='ref1')
+
+    # .state tests
+    def test_returns_files_instance_with_query_filtered_by_given_state(self):
+        # create many files
+        self._assert_filtered_by('state', ['state1', 'state2'], filter_by='state1')
+
+    # .path tests
+    def test_returns_files_instance_with_query_filtered_by_given_path(self):
+        self._assert_filtered_by('path', ['/path1', '/path2'], filter_by='/path1')
+
+    # .type tests
+    def test_returns_files_instance_with_query_filtered_by_given_type(self):
+        self._assert_filtered_by('type_', ['type1', 'type2'], files_method='type', filter_by='type1')
+
+    # TODO: check for in
+
+    # .group tests
+    @unittest.skip('group field removed from File. Delete Files.group too.')
+    def test_returns_files_instance_with_query_filtered_by_given_group(self):
+        self._assert_filtered_by('group', ['group1', 'group2'], filter_by='group1')
+
+    # .source_url tests
+    def test_returns_files_instance_with_query_filtered_by_given_source_url(self):
+        self._assert_filtered_by(
+            'source_url',
+            ['http://example.com/1', 'http://example.com/2'],
+            filter_by='http://example.com/1')
+
+    # .installed tests
+    def test_returns_files_instance_with_query_with_installed_files(self):
+        # create file for each value
+        f1 = FileFactory(type_=Files.TYPE.BUNDLE)
+        f2 = FileFactory(type_=Files.TYPE.PARTITION)
+        f3 = FileFactory(type_=Files.TYPE.SOURCE)
+        self.sqlite_db.session.commit()
+
+        # create Files instance with all files
+        query = self.sqlite_db.session.query(File)
+        all_files = Files(self.sqlite_db, query=query)
+
+        installed = all_files.installed
+        self.assertEquals(len(installed.all), 2)
+        self.assertIn(f1, installed.all)
+        self.assertIn(f2, installed.all)
+        self.assertNotIn(f3, installed.all)
+
+    # .new_file tests
+    def test_returns_new_file(self):
+        query = self.sqlite_db.session.query(File)
+        all_files = Files(self.sqlite_db, query=query)
+        self.assertEquals(len(all_files.all), 0)
+        new_f = all_files.new_file(
+            oid='oid1', path='/path1',
+            source_url='http://example.com', type_=Files.TYPE.BUNDLE)
+        self.assertIsInstance(new_f, File)
+        self.assertEquals(new_f.oid, 1)
+        self.assertEquals(new_f.path, '/path1')
+        self.assertEquals(new_f.type_, Files.TYPE.BUNDLE)
+        self.assertEquals(new_f.source_url, 'http://example.com')
+
+    # .merge tests
+    @unittest.skip('Where is Files().merge method? And why it is used in files.py?')
```

Comment:
Files.merge has been removed. It was part of a lot of wierdness to compensate for my lack of experience with Sqlalchemy, and the whole library.files was probably a bad idea